includes/User.php

```diff
@@ -2551,6 +2554,22 @@ public function setGlobalPreference($preference, $value) {
 	}
 
 	/**
+	 * @param array $preferences preferenceName->Value
+	 */
+	public function setGlobalPreferences( $preferences ) {
+		global $wgPreferencesUseService;
+		if ($wgPreferencesUseService) {
+			$this->sanitizePropertyArray($preferences);
+			$this->userPreferences()->setPreferences($this->mId, $preferences);
```

Comment:
Also maybe the logic from `User->setGlobalPreference` should be incorporated into `User->setGlobalPreferences` and `setGlobalPreference` should call `setGlobalPreferences`, since there is some additional logic/hooks that need to take place